Source/WebCore/css/parser/CSSPropertyParserConsumer+Color.cpp

```diff
@@ -820,96 +820,134 @@ static Color parseHSLParametersRaw(CSSParserTokenRange& range, ColorParserState&
     if (!whiteness)
-        return { };
+        return std::nullopt;
 
     auto blackness = whitenessAndBlacknessConsumer(args);
     if (!blackness)
-        return { };
+        return std::nullopt;
 
     auto alpha = alphaConsumer(args);
     if (!alpha)
-        return { };
+        return std::nullopt;
 
     if (!args.atEnd())
-        return { };
-
-    auto normalizedHue = WTF::switchOn(*hue,
-        [] (AngleRaw angle) { return CSSPrimitiveValue::computeDegrees(angle.type, angle.value); },
-        [] (NumberRaw number) { return number.value; },
-        [] (NoneRaw) { return std::numeric_limits<double>::quiet_NaN(); }
-    );
-    auto clampedWhiteness = WTF::switchOn(*whiteness,
-        [] (PercentRaw percent) { return std::clamp(percent.value, 0.0, 100.0); },
-        [] (NoneRaw) { return std::numeric_limits<double>::quiet_NaN(); }
-    );
-    auto clampedBlackness = WTF::switchOn(*blackness,
-        [] (PercentRaw percent) { return std::clamp(percent.value, 0.0, 100.0); },
-        [] (NoneRaw) { return std::numeric_limits<double>::quiet_NaN(); }
-    );
-
-    if (std::isnan(normalizedHue) || std::isnan(clampedWhiteness) || std::isnan(clampedBlackness) || std::isnan(*alpha)) {
-        auto [normalizedWhitness, normalizedBlackness] = normalizeClampedWhitenessBlacknessAllowingNone(clampedWhiteness, clampedBlackness);
-
-        // If any component uses "none", we store the value as a HWBA<float> to allow for storage of the special value as NaN.
-        return HWBA<float> { static_cast<float>(normalizedHue), static_cast<float>(normalizedWhitness), static_cast<float>(normalizedBlackness), static_cast<float>(*alpha) };
-    }
-
-    auto [normalizedWhitness, normalizedBlackness] = normalizeClampedWhitenessBlacknessDisallowingNone(clampedWhiteness, clampedBlackness);
-
-    if (normalizedHue < 0.0 || normalizedHue > 360.0) {
-        // If 'hue' is not in the [0, 360] range, we store the value as a HWBA<float> to allow for correct interpolation
-        // using the "specified" hue interpolation method.
-        return HWBA<float> { static_cast<float>(normalizedHue), static_cast<float>(normalizedWhitness), static_cast<float>(normalizedBlackness), static_cast<float>(*alpha) };
-    }
+        return std::nullopt;
 
-    // The explicit conversion to SRGBA<uint8_t> is an intentional performance optimization that allows storing the
-    // color with no extra allocation for an extended color object. This is permissible due to the historical requirement
-    // that HWBA colors serialize using the legacy color syntax (rgb()/rgba()) and historically have used the 8-bit rgba
-    // internal representation in engines.
-    return convertColor<SRGBA<uint8_t>>(HWBA<float> { static_cast<float>(normalizedHue), static_cast<float>(normalizedWhitness), static_cast<float>(normalizedBlackness), static_cast<float>(*alpha) });
+    return ParsedHWBA { *hue, *whiteness, *blackness, *alpha };
```

Comment:
Is there an alternate here where we just use two sets of braces instead of naming the type?